src/fondant/pipeline/pipeline.py

```diff
@@ -213,10 +235,13 @@ def _configure_caching_from_image_tag(
 
         return cache
 
-    @property
-    def dockerfile_path(self) -> t.Optional[Path]:
-        path = self.component_dir / "Dockerfile"
-        return path if path.exists() else None
+    def dockerfile_path(self, path: t.Union[str, Path]) -> t.Optional[Path]:
+        if self._is_custom_component(path):
+            component_dir = Path(path)
+        else:
+            component_dir = self._get_registry_path(str(path))
```

Comment:
I guess this change can be reverted? Although I would sill prefer it we can find a way to remove the `component_dir` argument from the `__init__` method.